### PR TITLE
Simplify telemetry structure; eliminate difference between critical and non-critical telemetry

### PR DESCRIPTION
**Pull request by dadlerj**
Proof of concept PR here. I'm failing in Buildkite and am struggling resolving some of them (I can't even seem to get prettier to run anymore, it's been too long). I would love to work with @andersonlauren and @ryphil to (1) make specific reductions to our telemetry as part of this or soon after, and (2) get eng support to fully clean up and remove the old config option.

## Test plan

* Run a local instance, set disableNonCriticalTelemetry to true
* See if any errors appear
* Confirm telemetry sent is identical regardless of the value of the configuration option above

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | -0.00% (-0.17 kb) | -0.00% (-0.17 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/remove-critical-telemetry/compare-report.html;tab=live_object) for a full comparison between the commits 5194127 and 01f5f64 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by akalia25** [comment]
hmm fixed the prettier issue buildkite was having, and now its showing a `Puppeteer tests chunk #9` , i'll try to look into this!

File: schema/site.schema.json [hunk 1]
```diff
@@ -706,7 +706,7 @@
       "group": "Security"
     },
     "disableNonCriticalTelemetry": {
-      "description": "Disable aggregated event counts from being sent to Sourcegraph.com via pings.",
+      "description": "DISABLED. Has no effect.",
```
Comment (unknwon) [inline schema/site.schema.json:709]:
```suggestion
      "description": "DEPRECATED. Has no effect.",
```

File: schema/schema.go [hunk 2]
```diff
@@ -2391,7 +2391,7 @@ type SiteConfiguration struct {
 	DisableAutoGitUpdates bool `json:"disableAutoGitUpdates,omitempty"`
 	// DisableFeedbackSurvey description: Disable the feedback survey
 	DisableFeedbackSurvey bool `json:"disableFeedbackSurvey,omitempty"`
-	// DisableNonCriticalTelemetry description: Disable aggregated event counts from being sent to Sourcegraph.com via pings.
+	// DisableNonCriticalTelemetry description: DISABLED. Has no effect.
```
Comment (unknwon) [inline schema/schema.go:2394]:
```suggestion
	// DisableNonCriticalTelemetry description: DEPRECATED. Has no effect.
```

**Review by unknwon — APPROVED** [review]

**Review by akalia25 — APPROVED** [review]

**Pull request merged**